### PR TITLE
Dockerized Jenkins

### PR DESCRIPTION
**Pull request by Feroz-Shaikh**

### CODE REVIEW

File: installscripts/wizard/scenarios/stack_with_bb_dockerized_jenkins.py [hunk 1]
```diff
@@ -1,8 +1,54 @@
...
+
+#Global variables
+HOME_FOLDER = os.path.expanduser("~")
+HOME_JAZZ_INSTALLER = HOME_FOLDER + "/jazz-installer/"
+HOME_INSTALL_SCRIPTS = HOME_JAZZ_INSTALLER + "installscripts/"
+JENKINS_DOCKER_PATH = HOME_INSTALL_SCRIPTS + "dockerfiles/jenkins/"
+TERRAFORM_FOLDER_PATH = HOME_INSTALL_SCRIPTS + "jazz-terraform-unix-noinstances"
+VARIABLES_TF_FILE = TERRAFORM_FOLDER_PATH + "variables.tf"
+JENKINS_PEM = HOME_FOLDER + "/jenkinskey.pem"
+BITBUCKET_PEM = HOME_FOLDER + "/bitbucketkey.pem"
```
Comment (SoorajSNair) [inline installscripts/wizard/scenarios/stack_with_bb_dockerized_jenkins.py]: we don't need a bitbucket per key here. Please remove.

File: installscripts/dockerfiles/jenkins/launch_jenkins_docker.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#Configuring and pulling the jenkins-docker from ECR
+sudo docker volume inspect jenkins-volume
+if [ $? == 0 ] ; then
+  sudo docker volume rm jenkins-volume
+fi
+
+sudo docker volume create jenkins-volume
+aws ecr get-login --no-include-email --region us-east-1 > ecr_login_script
+sudo bash ecr_login_script && rm -f ecr_login_script
+sudo docker pull 108206174331.dkr.ecr.us-east-1.amazonaws.com/jazz-oss:jenkins
```
Comment (suryajak) [inline installscripts/dockerfiles/jenkins/launch_jenkins_docker.sh]: We should parameterize this. When we move this to TMO account what happens. Should we use public dockerhub to store these images?

File: installscripts/dockerfiles/jenkins/launch_jenkins_docker.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+sudo usermod -aG docker $(whoami)
+
+#Configuring and pulling the jenkins-docker from ECR
+sudo docker volume inspect jenkins-volume
+if [ $? == 0 ] ; then
+  sudo docker volume rm jenkins-volume
+fi
+
+sudo docker volume create jenkins-volume
+aws ecr get-login --no-include-email --region us-east-1 > ecr_login_script
```
Comment (suryajak) [inline installscripts/dockerfiles/jenkins/launch_jenkins_docker.sh]: We are trying to generalize for jazz to work in any region, can we take this a parameter?

Comment (Feroz-Shaikh) [reply installscripts/dockerfiles/jenkins/launch_jenkins_docker.sh]: This is addressed in same PR.

File: installscripts/dockerfiles/jenkins/launch_jenkins_docker.sh (hunk 2 above)
Comment (Feroz-Shaikh) [reply installscripts/dockerfiles/jenkins/launch_jenkins_docker.sh]: @suryajak We need to test the login mechanism for docker-hub(password-encoding). Can we take this up as a future enhancement?

**Pull request merged**